File: packages/typescript/src/services/images.ts
import { BaseService } from './base.js';

export interface ImageGenerateOptions {
  model?: string;
  size?: string;
  n?: number;
  quality?: string;
}

export interface ImageResult {
  images?: Array<{ url?: string; b64_json?: string }>;
  jobId?: string;
}

export interface ImageJobStatus {
  jobId: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  result?: ImageResult;
}

export class ImagesService extends BaseService {
  async generate(prompt: string, options?: ImageGenerateOptions): Promise<ImageResult> {
    return this.request('POST', '/v1/images/generate', {
      body: { prompt, ...options },
    });
  }

  async generateAsync(prompt: string, options?: ImageGenerateOptions): Promise<{ jobId: string }> {
    return this.request('POST', '/v1/images/generate/queue', {
      body: { prompt, ...options },
    });
  }

  async getJob(jobId: string): Promise<ImageJobStatus> {
    return this.request('GET', `/v1/images/jobs/${encodeURIComponent(jobId)}`);
  }

  async getJobResult(jobId: string): Promise<ImageResult> {
    return this.request('GET', `/v1/images/jobs/${encodeURIComponent(jobId)}/result`);
  }

  async models(): Promise<{ models: Array<{ id: string; name: string }> }> {
    return this.request('GET', '/v1/images/models');
  }
}

File: packages/python/scalix_sdk/services/images.py
"""Image generation service."""

from __future__ import annotations

from typing import Any

from scalix_sdk.services.base import BaseService


class ImagesService(BaseService):
    async def generate(
        self,
        prompt: str,
        *,
        model: str | None = None,
        size: str | None = None,
        n: int | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"prompt": prompt}
        if model:
            body["model"] = model
        if size:
            body["size"] = size
        if n:
            body["n"] = n
        return await self._request("POST", "/v1/images/generate", json=body)

    async def generate_async(self, prompt: str, **kwargs: Any) -> dict[str, Any]:
        body: dict[str, Any] = {"prompt": prompt, **kwargs}
        return await self._request("POST", "/v1/images/generate/queue", json=body)

    async def get_job(self, job_id: str) -> dict[str, Any]:
        return await self._request("GET", f"/v1/images/jobs/{job_id}")

    async def get_job_result(self, job_id: str) -> dict[str, Any]:
        return await self._request("GET", f"/v1/images/jobs/{job_id}/result")

    async def models(self) -> dict[str, Any]:
        return await self._request("GET", "/v1/images/models")
